test/config_parser_SUITE.erl

```diff
@@ -2807,27 +2807,27 @@ registration_watchers(JidBins) ->
     #{<<"modules">> => #{<<"mod_register">> => Opts}}.
 
 mod_roster(_Config) ->
+    check_iqdisc_map(mod_roster),
+    check_module_defaults(mod_roster),
+    P = [modules, mod_roster],
     T = fun(Opts) -> #{<<"modules">> => #{<<"mod_roster">> => Opts}} end,
-    M = fun(Cfg) -> modopts(mod_roster, Cfg) end,
-
-    ?cfgh(M([{versioning, false}]),
-          T(#{<<"versioning">> => false})),
-    ?cfgh(M([{store_current_id, false}]),
-          T(#{<<"store_current_id">> => false})),
-    ?cfgh(M([{backend, mnesia}]),
-          T(#{<<"backend">> => <<"mnesia">>})),
-    ?cfgh(M([{bucket_type, <<"rosters">>}]),
-          T(#{<<"riak">> => #{<<"bucket_type">> => <<"rosters">>}})),
-    ?cfgh(M([{version_bucket_type, <<"roster_versions">>}]),
-          T(#{<<"riak">> => #{<<"version_bucket_type">> => <<"roster_versions">>}})),
+    ?cfgh(P ++ [versioning],  true,
+          T(#{<<"versioning">> => true})),
+    ?cfgh(P ++ [store_current_id], true,
+          T(#{<<"store_current_id">> => true})),
+    ?cfgh(P ++ [backend], rdbms,
+          T(#{<<"backend">> => <<"rdbms">>})),
```

Comment:
Could you check that Riak defaults are present if `backend` is set to `riak`?

Reply:
Please take a look if the test added is OK